spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/SchedulerTemplate.java

```diff
@@ -61,34 +61,76 @@
 	}
 
 	@Override
-	public void schedule(String scheduleName, String taskDefinitionName, Map<String, String> taskProperties, List<String> commandLineArgs) {
+	public void schedule(String scheduleName, String taskDefinitionName, Map<String, String> taskProperties, List<String> commandLineArgs, String platform) {
 		MultiValueMap<String, Object> values = new LinkedMultiValueMap<>();
 		values.add("scheduleName", scheduleName);
 		values.add("properties", DeploymentPropertiesUtils.format(taskProperties));
 		values.add("taskDefinitionName", taskDefinitionName);
 		values.add("arguments", commandLineArgs);
+		if(platform != null) {
```

Comment:
Instead of null, we can do StringUtils.hasText() or something related to avoid empty string.

Reply:
+1

Reply:
good catch.   Done